worksheet.rb

```diff
@@ -23,12 +57,218 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+# The date is stored in an array formatted as [YYYY, MM, DD].
+# Each element is an integer
+
+rides = [
+  {
+    driver_id: "DR0004",
+    date: [2016, 02, 03],
+    cost: 5.00,
+    rider_id: "RD0022",
+    rating: 5.0,
+  },
+
+  {
+    driver_id: "DR0001",
+    date: [2016, 02, 03],
+    cost: 10.00,
+    rider_id: "RD0003",
+    rating: 3.0,
+  },
+
+  {
+    driver_id: "DR0002",
+    date: [2016, 02, 03],
+    cost: 25.00,
+    rider_id: "RD0073",
+    rating: 5.0,
+  },
+
+  {
+    driver_id: "DR0001",
+    date: [2016, 02, 03],
+    cost: 30.00,
+    rider_id: "RD0015",
+    rating: 4.0,
+  },
+
+  {
+    driver_id: "DR0003",
+    date: [2016, 02, 04],
+    cost: 5.00,
+    rider_id: "RD0066",
+    rating: 5.0,
+  },
+
+  {
+    driver_id: "DR0004",
+    date: [2016, 02, 04],
+    cost: 10.00,
+    rider_id: "RD0022",
+    rating: 4.0,
+  },
+
+  {
+    driver_id: "DR0002",
+    date: [2016, 02, 04],
+    cost: 15.00,
+    rider_id: "RD0013",
+    rating: 1.0,
+  },
+
+  {
+    driver_id: "DR0003",
+    date: [2016, 02, 05],
+    cost: 50.00,
+    rider_id: "RD0003",
+    rating: 2.0,
+  },
+
+  {
+    driver_id: "DR0002",
+    date: [2016, 02, 05],
+    cost: 35.00,
+    rider_id: "RD0066",
+    rating: 3.0,
+  },
+
+  {
+    driver_id: "DR0004",
+    date: [2016, 02, 05],
+    cost: 20.00,
+    rider_id: "RD0073",
+    rating: 5.0,
+  },
+
+  {
+    driver_id: "DR0001",
+    date: [2016, 02, 05],
+    cost: 45.00,
+    rider_id: "RD0003",
+    rating: 2.0,
+  },
+]
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+
+# Make a hash to hold a hash for each driver's stats.
+
+drivers_data = {}
+
+rides.each do |ride|
+  if !drivers_data.has_key?(ride[:driver_id])
```

Comment:
I like that you set up this data structure here and then fill it in as you go. That makes the pattern of the code very clear.